ingestion/src/metadata/ingestion/source/database/snowflake/connection.py

```diff
@@ -132,16 +144,50 @@ def test_connection(
     Test connection. This can be executed either as part
     of a metadata workflow or during an Automation Workflow
     """
-
-    queries = {
-        "GetQueries": SNOWFLAKE_TEST_GET_QUERIES,
-        "GetDatabases": SNOWFLAKE_GET_DATABASES,
-        "GetTags": SNOWFLAKE_TEST_FETCH_TAG,
+    engine_wrapper = SnowflakeEngineWrapper(
+        service_connection=service_connection, engine=engine, is_use_executed=False
+    )
+    test_fn = {
+        "CheckAccess": partial(test_connection_engine_step, engine),
+        "GetDatabases": partial(
+            test_query, statement=SNOWFLAKE_GET_DATABASES, engine=engine
+        ),
+        "GetSchemas": partial(
+            execute_inspector_func, engine_wrapper, "get_schema_names"
+        ),
+        "GetTables": partial(execute_inspector_func, engine_wrapper, "get_table_names"),
+        "GetViews": partial(execute_inspector_func, engine_wrapper, "get_view_names"),
+        "GetQueries": partial(
+            test_query, statement=SNOWFLAKE_TEST_GET_QUERIES, engine=engine
+        ),
+        "GetTags": partial(
+            test_query, statement=SNOWFLAKE_TEST_FETCH_TAG, engine=engine
+        ),
     }
-    test_connection_db_common(
+
+    test_connection_steps(
         metadata=metadata,
-        engine=engine,
-        service_connection=service_connection,
+        test_fn=test_fn,
+        service_fqn=service_connection.type.value,
         automation_workflow=automation_workflow,
-        queries=queries,
     )
+
+
+def execute_inspector_func(engine_wrapper: SnowflakeEngineWrapper, func_name: str):
+    """
+    Method to test connection via inspector functions,
+    this function creates the inspector object and fetches
+    the function with name `func_name` and executes it
+    """
+    if (
+        not engine_wrapper.service_connection.database
+        and not engine_wrapper.is_use_executed
```

Comment:
to avoid unnecessary execution of use database query as it's okay to execute it only once for a session